sdk/tools/cabman/CCFDATAStorage.cxx

```diff
@@ -0,0 +1,167 @@
+/*
+ * PROJECT:     ReactOS cabinet manager
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     CCFDATAStorage class implementation for Linux/Unix
+ * COPYRIGHT:   Copyright 2017 Casper S. Hornstrup (chorns@users.sourceforge.net)
+ *              Copyright 2017 Colin Finck <mail@colinfinck.de>
+ *              Copyright 2018 Dmitry Bagdanov <dimbo_job@mail.ru>   
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#if !defined(_WIN32)
+#include <dirent.h>
+#endif
+
+#include "cabinet.h"
+#include "raw.h"
+#include "mszip.h"
+
+#if !defined(CAB_READ_ONLY)
+
+CCFDATAStorage::CCFDATAStorage()
+/*
+* FUNCTION: Default constructor
+*/
+{
+    FileCreated = false;
+}
+
+CCFDATAStorage::~CCFDATAStorage()
+/*
+* FUNCTION: Default destructor
+*/
+{
+    ASSERT(!FileCreated);
+}
+
+ULONG CCFDATAStorage::Create()
+/*
+* FUNCTION: Creates the file
+* ARGUMENTS:
+*     FileName = Pointer to name of file
+* RETURNS:
+*     Status of operation
+*/
+{
+    /*if (tmpnam(FullName) == NULL)*/
+    if ((FileHandle = tmpfile()) == NULL)
+        return CAB_STATUS_CANNOT_CREATE;
+    /*
+    FileHandle = fopen(FullName, "w+b");
+    if (FileHandle == NULL) {
+        DPRINT(MID_TRACE, ("ERROR '%i'.\n", errno));
+        return CAB_STATUS_CANNOT_CREATE;
+    }
+    */
+
+    FileCreated = true;
```

Comment:
I think you can remove FileCreated alltogether, and just check on `FileHandle != NULL` ?

Reply:
Removed in commit 753efbe694